examples/llm_compression/torch/qat_with_lora/main.py

```diff
@@ -0,0 +1,418 @@
+    model_to_eval = nncf.strip(model_to_eval, strip_format=StripFormat.DQ)
+    export_from_model(model_to_eval, ir_dir, device="cpu")
+    return OVModelForCausalLM.from_pretrained(
+        model_id=ir_dir,
+        trust_remote_code=True,
+        load_in_8bit=False,
+        compile=True,
+    )
+
+
+def get_argument_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(add_help=True)
+
+    # Model params
+    parser.add_argument(
+        "--pretrained",
+        type=str,
+        default="HuggingFaceTB/SmolLM-1.7B-Instruct",
+        help="The model id or path of a pretrained HF model configuration.",
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=str,
+        default="output",
+        help="Path to the directory for storing logs, tuning checkpoint, compressed model, validation references.",
+    )
+    parser.add_argument(
+        "--resume",
+        action="store_true",
+        help="Whether to start from previously saved checkpoint. If not specified or checkpoint does not exist, "
+        "start from scratch by post-training weight compression initializion.",
+    )
+
+    # Data params
+    parser.add_argument("--nsamples", type=int, default=1024, help="Number of training samples")
+    parser.add_argument("--seqlen", type=int, default=1024, help="Calibration data context length.")
+
+    # Training params
+    parser.add_argument(
+        "--lr",
+        type=float,
+        default=1e-4,
+        help="Learning rate for fine-tuning. "
+        "For larger models (over 2 billion parameters), a learning rate of 5e-4 is recommended.",
+    )
+    parser.add_argument("--epochs", type=int, default=10, help="Number of epochs.")
+    parser.add_argument("--batch_size", type=int, default=32, help="Size of training batch.")
+    parser.add_argument(
+        "--microbatch_size",
+        type=int,
+        default=8,
+        help="Size of each training microbatch. Gradients will be accumulated until the batch size is reached.",
+    )
+    return parser
+
+
+def main(argv) -> float:
+    """
+    Fine-tunes the specified model and returns the best validation similarity score.
+    """
+    parser = get_argument_parser()
+    args = parser.parse_args(argv)
+    assert torch.cuda.is_available()
+    set_seed(42)
+    device = "cuda"
+    torch_dtype = torch.bfloat16
+    compression_config = dict(
+        mode=CompressWeightsMode.INT4_ASYM, group_size=64, compression_format=CompressionFormat.FQ_LORA
+    )
+
+    # Configure output and log files.
+    output_dir = Path(args.output_dir)
+    tensorboard_dir = output_dir / "tb" / datetime.now().strftime("%D-%T")
+    last_dir = output_dir / "last"
+    best_dir = output_dir / "best"
+    for path in [output_dir, tensorboard_dir, last_dir, best_dir]:
+        if not args.resume:
+            shutil.rmtree(path, ignore_errors=True)
+        path.mkdir(exist_ok=True, parents=True)
```

Comment:
Creating folders better made inside `save_checkpoint` function

Reply:
There are different type of folders, e.g. tensorboard_dir. Not all go through `save_checkpoint` function. 
I prefer to have everything initialized in the very beginning and don't care about folders anymore inside the sample.